bootstrap/kubeadm/api/v1beta1/kubeadm_types.go

```diff
@@ -53,6 +53,22 @@ type InitConfiguration struct {
 	// fails you may set the desired value here.
 	// +optional
 	LocalAPIEndpoint APIEndpoint `json:"localAPIEndpoint,omitempty"`
+
+	// CertificateKey sets the key with which certificates and keys are encrypted prior to being uploaded in
+	// a secret in the cluster during the uploadcerts init phase.
+	// +optional
+	CertificateKey string `json:"certificateKey,omitempty"`
+
+	// SkipPhases is a list of phases to skip during command execution.
+	// The list of phases can be obtained with the "kubeadm init --help" command.
+	// The flag "--skip-phases" takes precedence over this field.
```

Comment:
```suggestion
	// SkipPhases is a list of phases to skip during command execution.
	// The list of phases can be obtained with the "kubeadm init --help" command.
```
Given that users are not allowed to set kubeadm flags via the CABPK API.

Also, we should document that this flag only works for kubernetes >= 1.22 (please double-check this), given the proposed implementation relies only on the kubeadm config API.
(same for join configuration)

Reply:
Given the discussion in https://github.com/kubernetes-sigs/cluster-api/issues/5357 I wonder if it would be a good idea to address adding `SkipPhases` and `Patches` in separate PRs.

Reply:
Potentially we could start with silently only supporting newer versions and then adding some validation in a follow-up issue/PR?

Reply:
@fabriziopandini WDYT? (forgot the mention before)

Reply:
I gave for granted that we are documenting those fields would work only from a given version, and fail silently for older ones, but open to discuss alternatives.

Reply:
Definitely fine for me to document it on the field accordingly and ignore silently on older Kubernetes versions.